src/uu/unexpand/src/unexpand.rs

```diff
@@ -35,37 +35,125 @@ enum ParseError {
 
 impl UError for ParseError {}
 
-fn tabstops_parse(s: &str) -> Result<Vec<usize>, ParseError> {
+fn parse_increment_syntax(word: &str, nums: &[usize]) -> Result<usize, ParseError> {
+    if nums.is_empty() {
+        return Err(ParseError::InvalidCharacter("+".to_string()));
+    }
+    match word.parse::<usize>() {
+        Ok(num) => {
+            if num == 0 {
+                return Err(ParseError::TabSizeCannotBeZero);
+            }
```

Comment:
I would handle this case in its own arm: it makes the `match` more compact and allows you to get rid of the `return`.